ring.go

```diff
@@ -76,6 +80,10 @@ type RingOptions struct {
 }
 
 func (opt *RingOptions) init() {
+	if opt.Passwords == nil {
+		opt.Passwords = make(map[string]string)
```

Comment:
This can be removed. No need for an allocation - reading from nil map is fine.

Reply:
Done!